### PR TITLE
SetServiceUrl

### PR DESCRIPTION
**Pull request by mediumTaj**
This pull request refactors SetEndpoint to SetServiceUrl.

BREAKING CHANGE: Use SetServiceUrl to set the service endpoint rather than SetEndpoint

### CODE REVIEW

**Comment by mediumTaj** [comment]
Not certain why the NoAuth test shows as deleted. Its in the repo
https://github.com/IBM/dotnet-sdk-core/blob/7504-set-service-url/test/IBM.Cloud.SDK.Core.Tests/NoAuthTests.cs

Edit: On second thought `NoauthTests.cs` shows as deleted. Probably an issue with case sensitive file system.

File: src/IBM.Cloud.SDK.Core/Service/IBMService.cs [hunk 1]
```diff
@@ -60,14 +60,14 @@ protected string Endpoint
...
-        protected IBMService(string serviceName, string url, IClient httpClient)
+        protected IBMService(string serviceName, string serviceUrl, IClient httpClient)
         {
             ServiceName = serviceName;
             Client = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
             authenticator = new NoAuthAuthenticator();
 
             if (!string.IsNullOrEmpty(Endpoint))
-                Endpoint = url;
+                Endpoint = serviceUrl;
```
Comment (padamstx) [inline src/IBM.Cloud.SDK.Core/Service/IBMService.cs:70]: Is this the correct conditional check here???
I read this as... "If the 'Endpoint' value currently set on this IBMService instance is not null or empty, then assign the serviceUrl value to it", which doesn't make sense.

In fact, I'm not quite sure we want to pass in the serviceUrl parameter to the constructor anyway.
Why not let the IBMService instance be constructed with a null Endpoint/ServiceURL value, then let the
generator decide if it wants to set the default service url by generating the code to do that.
So, I guess I'm suggesting we simply omit the "serviceUrl" parameter to this constructor altogether.

**Review by padamstx — CHANGES_REQUESTED** [review]
I think there's one detail we need to resolve

File: src/IBM.Cloud.SDK.Core/Service/IBMService.cs (hunk 1 above)
Comment (mediumTaj) [reply src/IBM.Cloud.SDK.Core/Service/IBMService.cs:70]: This PR is _only_ for refactoring SetEndpoint to SetServiceUrl. I'm working the core changes for url for another PR.

**Pull request merged**